ftw/simplelayout/utils.py

```diff
@@ -14,3 +19,17 @@ def get_block_html(block):
                                    IBlockProperties)
     viewname = properties.get_current_view_name()
     return block.restrictedTraverse(viewname)()
+
+
+def get_block_types():
+    platform = getSite()
+    types_tool = getToolByName(platform, 'portal_types')
+    typeids = types_tool.objectIds()
+    typeids.sort()
+    for type_id in typeids:
+        dx_fti = queryUtility(IDexterityFTI, name=type_id)
```

Comment:
Testing the actual FTI for `plone.dexterity.interfaces.IDexterityFTI` would be more efficient, since we would not have to make a utility lookup for each FTI each time.
Something like this should work:
```python
block_ftis = filter(IDexterityFTI.providedBy, types_tool.objectValues())
return map(methodcaller('getId'), block_ftis)
```